Add a dependabot configuration

bickelj opened the pull request:
Hopefully dependabot will "just work" but the presence of https://github.com/dependabot/dependabot-core/issues/390 suggests it will not. This repository doesn't have a `Dockerfile`, only a `compose.yml`.

port19x commented:
> Hopefully dependabot will "just work" but the presence of [dependabot/dependabot-core#390](https://github.com/dependabot/dependabot-core/issues/390) suggests it will not. This repository doesn't have a `Dockerfile`, only a `compose.yml`.

It won't work btw

slifty commented:
@bickelj Should we explore leveraging a Dockerfile in order to get dependabot support (either in this PR or by splitting the docker based dependabot stuff into a new PR)

slifty reviewed (changes requested):
Since it sounds like the `docker` portion of this change won't do anything in the current setup, let's either (A) remove that portion from this PR or (B) also adjust our docker configuration approach so that it uses a Dockerfile.

I'd probably lean towards (A) since that way we at least start getting github-actuion dependabot updates while we work on (B) in another PR.

port19x commented:
https://github.com/sbe-arg/simple-compose-service-updates might suit your use-case as well. Not sure how you guys would move a compose stack into a Dockerfile

bickelj commented:
@port19x is right: trying to shoehorn a compose script into a Dockerfile is intractable because compose operates at a level higher than already-built docker images.

@slifty I'm still inclined to merge the docker portion along with it. Suppose https://github.com/dependabot/dependabot-core/issues/390 gets fixed. I'd rather dependabot just start working than have to wait for notification of that issue getting fixed and then going back and updating our configuration. But I'm not strongly inclined either way.

slifty commented:
@bickelj that's fine!

For now is this PR actionable?  Let's either update it so we can merge it, or close it with the intent to re-open when conditions have resolved.

bickelj committed:
Add a dependabot configuration

Hopefully dependabot will "just work" but the presence of
https://github.com/dependabot/dependabot-core/issues/390 suggests it
will not. This repository doesn't have a `Dockerfile`, only a
`compose.yml`.

As of this commit it is not expected that dependabot will immediately
work for Docker images, but we hope the above issue will be fixed and
at that point dependabot will report new images.

bickelj commented:
@slifty Does it look OK? I rebased on main and added a comment to the commit message indicating why I kept the Docker part in.

bickelj commented:
@port19x Thank you for confirming the suspicion that Docker Compose isn't yet working and also for the suggested workaround. Much appreciated!

slifty reviewed (approved):
Sorry it took so long for me to approve this (I had misunderstood the conversation on the PR!)

Thank you @bickelj for the work and for @port19x for weighing in with some context and guidance.
